src/USER-CAC/pair_CAC.h

```diff
@@ -0,0 +1,200 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifdef PAIR_CLASS
+
+PairStyle(CAC,PairCAC)
+
+#else
+
+#ifndef LMP_PAIR_CAC_H
+#define LMP_PAIR_CAC_H
+
+
+#include "pair.h"
+#include <vector>
+#include <stdint.h>
+#include "memory.h"
+#include "asa_user.h"
```

Comment:
this should not be included here, but include it in the implementation.

Reply:
Im a bit at loss as to how to refactor asa_user.h out of this header file given that the functions myvalue and mygrad depend on such a definition. Moving it all to another class that's forward declared then requires moving a large volume of data, specific to the objective of each calling class such as deciding on reneighbors vs. point to surface minimization; this is due to the requirement of defining myvalue and mygrad for the asa cg code to call. The interface between the function calls to the style implementation and the asa cg code is not simple since it relies on member function pointers with structure inputs.

Reply:
I think I have an idea but it'll probably take a while to untangle the spider web. On the bright side it should also make adding new callers to the asa min algorithm easier in the future if it works as intended.